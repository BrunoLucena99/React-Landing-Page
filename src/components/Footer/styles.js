import styled from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  text-align: center;
  border-top: ${({ theme }) => `0.1rem solid ${theme.colors.mediumGray}`};

  a {
    color: inherit;
    text-decoration: none;
  }

  ${TextComponent} {
    font-size: ${({ theme }) => theme.font.sizes.small};
  }

  ${SectionContainer} {
    padding-top: 0;
    padding-bottom: 0;
  }
`;
