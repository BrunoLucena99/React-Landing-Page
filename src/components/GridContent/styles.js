import styled from 'styled-components';

export const Container = styled.div`
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
`;

export const HTMLContainer = styled.div`
  margin: ${({ theme }) => theme.spacings.xhuge} 0;
`;
