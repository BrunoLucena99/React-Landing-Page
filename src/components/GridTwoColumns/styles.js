import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.xlarge};

    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
      text-align: center;
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
