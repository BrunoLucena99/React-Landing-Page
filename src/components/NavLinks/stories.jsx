import NavLinks from '.';
import links from '../../mocks/links';

export default {
  title: 'NavLinks',
  component: NavLinks,
  args: {
    links,
  },
  argTypes: {
    links: { type: '' },
  },
};

export const Template = (args) => (
  <div>
    <NavLinks {...args} />
  </div>
);
