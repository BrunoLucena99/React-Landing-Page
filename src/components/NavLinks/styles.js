import styled, { css } from 'styled-components';

export const Container = styled.nav`
  display: flex;
  flex-flow: row wrap;

  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      flex-flow: column nowrap;
      align-items: center;
      justify-content: center;
    }
  `}
`;
