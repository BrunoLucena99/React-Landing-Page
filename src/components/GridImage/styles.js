import styled from 'styled-components';
import { Title as HeadingTitle } from '../Heading/styles';

export const Container = styled.div``;

export const Grid = styled.div`
  display: grid;
  gap: ${({ theme }) => theme.spacings.large};
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  margin-top: ${({ theme }) => theme.spacings.xhuge};
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
