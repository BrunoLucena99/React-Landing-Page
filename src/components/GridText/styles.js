import styled from 'styled-components';
import { Title as HeadingTitle } from '../Heading/styles';

export const Container = styled.div``;

export const Grid = styled.div`
  counter-reset: grid-counter;
  display: grid;
  gap: ${({ theme }) => theme.spacings.large};
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  margin-top: ${({ theme }) => theme.spacings.xhuge};
`;

export const GridElement = styled.div`
  ${HeadingTitle} {
    position: relative;
    left: 5rem;
  }

  ${HeadingTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
`;
