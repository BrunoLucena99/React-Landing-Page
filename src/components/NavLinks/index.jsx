import P from 'prop-types';
import MenuLink from '../MenuLink';
import { Container } from './styles';

const NavLinks = ({ links = [] }) => {
  return (
    <Container>
      {links.map((link) => (
        <MenuLink key={link.href} {...link} />
      ))}
    </Container>
  );
};

NavLinks.propTypes = {
  links: P.arrayOf(
    P.shape({
      children: P.node.isRequired,
      href: P.string.isRequired,
      newTab: P.bool,
    }),
  ),
};

export default NavLinks;
