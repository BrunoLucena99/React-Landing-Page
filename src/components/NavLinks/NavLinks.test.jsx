import { renderTheme } from '../../styles/RenderTheme';
import NavLinks from '.';
import { screen } from '@testing-library/react';
import links from '../../mocks/links';
import { theme } from '../../styles/theme';

describe('<NavLinks />', () => {
  it('should render correctly', () => {
    renderTheme(<NavLinks links={links} />);
    expect(screen.getAllByRole('link')).toHaveLength(links.length);
  });

  it('should flex-flow row wrap on normal devices', () => {
    const { container } = renderTheme(<NavLinks links={links} />);
    expect(container.firstChild).toHaveStyle({
      'flex-flow': 'row wrap',
    });
  });

  it('should flex-flow column on mobile devices', () => {
    const { container } = renderTheme(<NavLinks links={links} />);
    expect(container.firstChild).toHaveStyleRule('flex-flow', 'column nowrap', {
      media: theme.media.lteMedium,
    });
  });

  it('should render without links', () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByRole('link')).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={links} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
