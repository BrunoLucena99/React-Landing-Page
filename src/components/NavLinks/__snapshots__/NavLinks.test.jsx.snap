// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c1 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="#inicio"
    target="_self"
  >
    Inicio
  </a>
  <a
    class="c1"
    href="#Link 2"
    target="_self"
  >
    Link 2
  </a>
  <a
    class="c1"
    href="#Link 3"
    target="_self"
  >
    Link 3
  </a>
  <a
    class="c1"
    href="#Link 4"
    target="_self"
  >
    Link 4
  </a>
  <a
    class="c1"
    href="#Link 5"
    target="_self"
  >
    Link 5
  </a>
  <a
    class="c1"
    href="#Link 6"
    target="_self"
  >
    Link 6
  </a>
  <a
    class="c1"
    href="#Link7"
    target="_self"
  >
    Link7
  </a>
  <a
    class="c1"
    href="#Link8"
    target="_self"
  >
    Link8
  </a>
  <a
    class="c1"
    href="#Link9"
    target="_self"
  >
    Link9
  </a>
  <a
    class="c1"
    href="#Link 10"
    target="_self"
  >
    Link 10
  </a>
</nav>
`;
